### PR TITLE
Miscellaneous patches

### PR DESCRIPTION
**Pull request by luckysori**
The motivation for 38c54c53ec8d8525197a4c593e08df4781617867, ab73ccedbf2123d54395e4579807ac331e819d64 and 40bd0c45ac5688e69b646a8b30255ac290202ab6 is to be able to use the async background processor without problems. These bug fixes are part of 0.0.115.

40bd0c45ac5688e69b646a8b30255ac290202ab6 is useful because the disconnects are important for debugging at the moment, but TRACE-level logs are too noisy in that module.

d6e4ba8849d99ada8be2d4503c869c96e37ee284 is not so important so we could drop it, but it's also very simple so it can't hurt.

b4e4eb030470aa97a597f3e0f89ea40e56e4b189 might be the most controversial, but we did run into a related `unwrap` a few times, so it seemed like a good idea at the time.

### CODE REVIEW

File: lightning/src/util/enforcing_trait_impls.rs [hunk 1]
```diff
@@ -259,13 +270,13 @@ impl Writeable for EnforcingSigner {
 impl EnforcingSigner {
 	fn verify_counterparty_commitment_tx<'a, T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &'a CommitmentTransaction, secp_ctx: &Secp256k1<T>) -> TrustedCommitmentTransaction<'a> {
 		commitment_tx.verify(&self.inner.get_channel_parameters().as_counterparty_broadcastable(),
-		                     self.inner.counterparty_pubkeys(), self.inner.pubkeys(), secp_ctx)
```
Comment (Tibo-lg) [inline lightning/src/util/enforcing_trait_impls.rs]: nit: maybe ok to remove?

File: lightning/src/util/enforcing_trait_impls.rs [hunk 2]
```diff
@@ -259,13 +270,13 @@ impl Writeable for EnforcingSigner {
...
+				     self.inner.counterparty_pubkeys(), self.inner.pubkeys(), secp_ctx)
 			.expect("derived different per-tx keys or built transaction")
 	}
 
 	fn verify_holder_commitment_tx<'a, T: secp256k1::Signing + secp256k1::Verification>(&self, commitment_tx: &'a CommitmentTransaction, secp_ctx: &Secp256k1<T>) -> TrustedCommitmentTransaction<'a> {
 		commitment_tx.verify(&self.inner.get_channel_parameters().as_holder_broadcastable(),
-		                     self.inner.pubkeys(), self.inner.counterparty_pubkeys(), secp_ctx)
```
Comment (Tibo-lg) [inline lightning/src/util/enforcing_trait_impls.rs]: ditto

File: lightning-background-processor/src/lib.rs [hunk 3]
```diff
@@ -788,7 +791,10 @@ mod tests {
 
 			if key == "network_graph" {
 				if let Some(sender) = &self.graph_persistence_notifier {
-					sender.send(()).unwrap();
+					match sender.send(()) {
```
Comment (Tibo-lg) [inline lightning-background-processor/src/lib.rs:794]: This is in only in tests, just wanted to check that the intention was to fix test related things as the commit name didn't imply this.

File: lightning/src/util/events.rs [hunk 4]
```diff
@@ -462,7 +462,9 @@ pub enum BumpTransactionEvent {
 	/// [`BaseSign::sign_holder_htlc_transaction`]: crate::chain::keysinterface::BaseSign::sign_holder_htlc_transaction
 	/// [`HTLCDescriptor::tx_input_witness`]: HTLCDescriptor::tx_input_witness
 	HTLCResolution {
+		///
```
Comment (Tibo-lg) [inline lightning/src/util/events.rs]: are those empty comments necessary?

**Review by Tibo-lg — COMMENTED** [review]
just a couple of nits, otherwise LGTM

File: lightning-background-processor/src/lib.rs (hunk 3 above)
Comment (luckysori) [reply lightning-background-processor/src/lib.rs:794]: That's true!

I cherry-picked this from `lightningdevkit/rust-lightning` and I guess these lines were included too. That's why `tnull` is a co-author here. I would just leave them in, but the rest of the patch is what is actually important.

Btw, these changes will be there when we update to 0.0.115, as well as some of the other changes in this PR.

File: lightning/src/util/enforcing_trait_impls.rs (hunk 1 above)
Comment (luckysori) [reply lightning/src/util/enforcing_trait_impls.rs]: Ah, fair. 

This is just my editor fixing the inconsistent use of spaces and tabs here. I think `rust-lightning` uses tabs, but it's not enforced.

File: lightning/src/util/events.rs (hunk 4 above)
Comment (luckysori) [reply lightning/src/util/events.rs]: Fixed!

File: lightning/src/util/enforcing_trait_impls.rs (hunk 1 above)
Comment (luckysori) [reply lightning/src/util/enforcing_trait_impls.rs]: I removed it!

File: lightning/src/util/enforcing_trait_impls.rs (hunk 2 above)
Comment (luckysori) [reply lightning/src/util/enforcing_trait_impls.rs]: Removed too.

**Review by Tibo-lg — APPROVED** [review]

**Pull request merged**